mmeval/metrics/lvis_detection.py

```diff
@@ -0,0 +1,297 @@
+                lvis_eval.params.max_dets = self.proposal_nums
+                lvis_eval.evaluate()
+                lvis_eval.accumulate()
+                lvis_eval.summarize()
+                if metric_items is None:
+                    metric_items = [
+                        f'AR@{self.proposal_nums}',
+                        f'ARs@{self.proposal_nums}',
+                        f'ARm@{self.proposal_nums}',
+                        f'ARl@{self.proposal_nums}'
+                    ]
+                for k, v in lvis_eval.get_results().items():
+                    if k in metric_items:
+                        val = float(f'{float(v):.3f}')
+                        eval_results[k] = val
+
+            else:
+                lvis_eval.evaluate()
+                lvis_eval.accumulate()
+                lvis_eval.summarize()
+                lvis_results = lvis_eval.get_results()
+
+                if metric_items is None:
+                    metric_items = [
+                        'AP', 'AP50', 'AP75', 'APs', 'APm', 'APl', 'APr',
+                        'APc', 'APf'
+                    ]
+
+                results_list = []
+                for metric_item, v in lvis_results.items():
+                    if metric_item in metric_items:
+                        key = f'{metric}_{metric_item}'
+                        val = float(v)
+                        results_list.append(f'{round(val * 100, 2)}')
+                        eval_results[key] = val
+                eval_results[f'{metric}_result'] = results_list
+
+                if self.classwise:  # Compute per-category AP
+                    # Compute per-category AP
+                    # from https://github.com/facebookresearch/detectron2/
+                    precisions = lvis_eval.eval['precision']
+                    # precision: (iou, recall, cls, area range, max dets)
+                    assert len(self.cat_ids) == precisions.shape[2]
+
+                    results_per_category = []
+                    for idx, catId in enumerate(self.cat_ids):
+                        # area range index 0: all area ranges
+                        # max dets index -1: typically 100 per image
+                        # the dimensions of precisions are
+                        # [num_thrs, num_recalls, num_cats, num_area_rngs]
+                        nm = self._lvis_api.load_cats([catId])[0]
+                        precision = precisions[:, :, idx, 0]
+                        precision = precision[precision > -1]
+                        if precision.size:
+                            ap = np.mean(precision)
+                        else:
+                            ap = float('nan')
+                        results_per_category.append(
+                            (f'{nm["name"]}', f'{round(ap * 100, 2)}'))
+                        eval_results[f'{metric}_{nm["name"]}_precision'] = ap
+
+                    eval_results[f'{metric}_classwise_result'] = \
```

Comment:
If we have logger in Metrics, we can print tables inside the metric. Hence, the output can simply be dict(str, float). e.g.: https://github.com/open-mmlab/mmeval/pull/106/files#diff-ed3e61b673a35151c770f53e4be0ed88c273eec4134166e1775670fa0d114aa2R617